### ISSUE TITLE
Gulp plugin, exposed API

### ISSUE THREAD

**Issue by pascalduez** [SassDoc/sassdoc#97]
I'm currently working on a Gulp plugin, and wrapping my head around streams and stuff.
I came to the conclusion that it is actually not relevant.

Gulp is a build tool, it streams, process and returns files.
In sassdoc (the lib) case, all that work is already done by the API.
We could redo a good part of it in the plugin, but that sounds a bit lame. Plus it would be more subject to break in case of API changes.

Still, the sassdoc API could be a bit more modular and welcoming for external projects.
For instance, export the assets dir path.
But he main concern is about the `getData` function which would benefit to be decoupled.
Basically, decouple IO based operation, and data massaging ones.

One solution for the plugin though, is to build a so called "Gulp friendly" plugin, not using streams, but making sassdoc inclusion in a Gulp workflow a bit easier.
